lib/puppet/provider/gpo/lgpo.rb

```diff
@@ -138,4 +140,18 @@ def set_value(val)
     lgpo(*lgpo_args)
     File.delete(out_polfile_path)
   end
+
+  def remove_key(key, scope)
+    pol_file = "C:\\Windows\\System32\\GroupPolicy\\#{scope.capitalize}\\Registry.pol"
+    out_file_path = File.join(Puppet[:vardir], 'lgpo_import.txt')
+    gpos = lgpo('/parse', '/q', "/#{scope[0]}", pol_file)
+    gpos.split("\n\n").reject { |l| l.start_with? ';' }.each do |g|
+      split_g = g.split("\n")
+      unless split_g[1] == key
+        File.open(out_file_path, 'a') do |out_file|
```

Comment:
I would rather open the file outside the loop to avoid opening/closing for every line.